pages/transactions/[id].js

```diff
@@ -119,25 +135,21 @@ export default function TransactionDetailsPage({
         />
       </Modal>
 
-      {successMessage && (
-        <StyledSuccessMessage>{successMessage}</StyledSuccessMessage>
-      )}
-
       <StyledTransactionDetails>
         <StyledTitle>Transaction Details</StyledTitle>
 
-        {isDeleting ? (
+        {isDeletingTransaction ? (
           <StyledConfirmActionContainer>
             <StyledCancelButton
               type="button"
-              onClick={handleCancelDeleteDialogue}
+              onClick={() => setIsDeletingTransaction(false)}
             >
               Cancel
             </StyledCancelButton>
             <StyledConfirmButton
               type="button"
               onClick={() => {
-                handleConfirmDelete(transactionDetails);
+                handleConfirmDeleteTransaction();
                 router.push("/");
```

Comment:
there is already a router.push inside the handler function. you can remove it here